exercises/chapter6/test/no-peeking/Solutions.purs

```diff
@@ -107,23 +105,14 @@ instance actionArray :: Action m a => Action m (Array a) where
 newtype Self m
   = Self m
 
--- Why is Monoid constraint required here?
--- Seems like this is already specified by Action class
---instance actionSelf :: Action m (Self m) where
 instance actionSelf :: Monoid m => Action m (Self m) where
   act m1 (Self m2) = Self (m1 <> m2)
 
-instance eqSelf :: Eq m => Eq (Self m) where
-  eq (Self m1) (Self m2) = m1 == m2
-
-instance showSelf :: Show m => Show (Self m) where
-  show (Self m) = "Self " <> show m
-
-instance semigroupSelf :: Semigroup m => Semigroup (Self m) where
-  append (Self a) (Self b) = Self (a <> b)
-
-instance monoidSelf :: Monoid m => Monoid (Self m) where
-  mempty = Self mempty
+-- These may also be written manualy
+derive newtype instance showSelf :: Show m => Show (Self m)
+derive newtype instance eqSelf :: Eq m => Eq (Self m)
+derive newtype instance semigroupSelf :: Semigroup m => Semigroup (Self m)
+derive newtype instance monoidSelf :: Monoid m => Monoid (Self m)
 
 instance repeatActionMultSelf :: Action (Self Multiply) Int where
```

Comment:
This might not be the best instance name (doesn't follow the earlier convention of just squishing everything together), although this doesn't really matter.